src/popup/components/Protocol/EnsureUserSavedMnemonic.tsx

```diff
@@ -47,28 +55,56 @@ const WordButton = (props: {
   onClick: () => void;
 }) => {
   return (
-    <Button disabled={!props.isEnabled} onClick={props.onClick}>
+    <Button
+      className={(() => {
+        return !props.isEnabled ? "ClickedWord" : "";
+      })()}
+      onClick={props.onClick}
+    >
       {props.word}
     </Button>
   );
 };
 
-export function EnsureUserSavedMnemonic(props: { onComplete: () => void }) {
-  const [buttons, setButtons] = React.useState<CheckButton[]>([]);
-  const [mnemonicArr, setMnemonicArr] = React.useState<string[]>([]);
+const TextField = (props: { input: string }) => {
+  return <TextArea defaultValue={props.input} />;
+};
 
-  const [counter, setCounter] = React.useState(0);
+export function EnsureUserSavedMnemonic(props: { onComplete: () => void }) {
+  const [pageOverride, setPageOverride] = useState<JSX.Element | null>(null);
+  const [buttons, setButtons] = useState<CheckButton[]>([]);
+  const [success, setSuccess] = useState(false);
+  const [mnemonic, setMnemonic] = useState("");
+  const [currentInput, setCurrentInput] = useState<string[]>([]);
 
   useEffect(() => {
     async function getMnemonic() {
-      const mnemonic = await getProtocolOptIn().getMnemonic();
-      setMnemonicArr((mnemonic as string).split(" "));
-      const sortedMnemonic = (mnemonic as string).split(" ").sort();
+      setMnemonic((await getProtocolOptIn().getMnemonic()) as string);
+      const sortedMnemonic = mnemonic.split(" ").sort();
       setButtons(sortedMnemonic.map((w) => ({ word: w, isEnabled: true })));
+      setPageOverride(
+        <SetupSuccess
+          mnemonic={mnemonic as string}
+          onContinue={() => setPageOverride(null)}
+        />,
+      );
     }
 
     getMnemonic();
-  }, []);
+  }, [mnemonic]);
```

Comment:
This only should need to run once. Should have `[]` as the variables to depend on.